Src/CSharpier.Cli/Options/ConfigurationFileOptions.cs

```diff
@@ -16,128 +12,70 @@ internal class ConfigurationFileOptions
     [JsonConverter(typeof(CaseInsensitiveEnumConverter<EndOfLine>))]
     public EndOfLine EndOfLine { get; init; }
 
-    private static readonly string[] validExtensions = { ".csharpierrc", ".json", ".yml", ".yaml" };
+    public Override[] Overrides { get; init; } = [];
 
-    internal static PrinterOptions CreatePrinterOptionsFromPath(
-        string configPath,
-        IFileSystem fileSystem,
-        ILogger logger
-    )
+    public PrinterOptions? ConvertToPrinterOptions(string filePath)
     {
-        var configurationFileOptions = Create(configPath, fileSystem, logger);
-
-        return ConvertToPrinterOptions(configurationFileOptions);
-    }
-
-    internal static PrinterOptions ConvertToPrinterOptions(
-        ConfigurationFileOptions configurationFileOptions
-    )
-    {
-        return new PrinterOptions
-        {
-            TabWidth = configurationFileOptions.TabWidth,
-            UseTabs = configurationFileOptions.UseTabs,
-            Width = configurationFileOptions.PrintWidth,
-            EndOfLine = configurationFileOptions.EndOfLine
-        };
-    }
-
-    /// <summary>Finds all configs above the given directory as well as within the subtree of this directory</summary>
-    internal static List<CSharpierConfigData> FindForDirectoryName(
-        string directoryName,
-        IFileSystem fileSystem,
-        ILogger logger,
-        bool limitEditorConfigSearch
-    )
-    {
-        var results = new List<CSharpierConfigData>();
-        var directoryInfo = fileSystem.DirectoryInfo.New(directoryName);
-
-        var filesByDirectory = directoryInfo
-            .EnumerateFiles(
-                ".csharpierrc*",
-                limitEditorConfigSearch
-                    ? SearchOption.TopDirectoryOnly
-                    : SearchOption.AllDirectories
-            )
-            .GroupBy(o => o.DirectoryName);
-
-        foreach (var group in filesByDirectory)
+        DebugLogger.Log("finding options for " + filePath);
+        var matchingOverride = this.Overrides.LastOrDefault(o => o.IsMatch(filePath));
+        if (matchingOverride is not null)
         {
-            var firstFile = group
-                .Where(o => validExtensions.Contains(o.Extension, StringComparer.OrdinalIgnoreCase))
-                .MinBy(o => o.Extension);
-
-            if (firstFile != null)
+            return new PrinterOptions
             {
-                results.Add(
-                    new CSharpierConfigData(
-                        firstFile.DirectoryName!,
-                        Create(firstFile.FullName, fileSystem, logger)
-                    )
-                );
-            }
+                TabWidth = matchingOverride.TabWidth,
+                UseTabs = matchingOverride.UseTabs,
```

Comment:
out of curiosity, if the matching override contains a `TabWidth` set, but not `UseTabs`, will it just go the default value of `UseTabs`? Or will it pick the value from the global printer options that have been set elsewhere?

I.e. is it okay that you're not doing
```csharp
TabWidth = matchingOverride.TabWidth ?? this.TabWith,
/* same for the rest of the properties */
```

You'll also have to make the relevant properties in `Override` nullable of course.

Reply:
It appears that it is poorly named. I ran across these TODOs when noodling on renaming it. Apparently I had the idea already. Gonna just sneak it in with this PR, it seems easy enough to do.

```
// TODO rename this to PrintingContext
// TODO and rename PrinterOptions.TabWidth to PrinterOptions.IndentSize
internal class FormattingContext
```

The configuration file does use the `tabWidth` name, which seems to have come from prettier. Changing that seems not worth the headache.